lang/cs/Org.Apache.REEF.Common/Evaluator/DriverInformation.cs

```diff
@@ -89,48 +90,73 @@ public string NameServerId
 
         public static DriverInformation GetDriverInformationFromHttp(Uri queryUri)
         {
-            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(queryUri);
-            request.AllowAutoRedirect = false;
-            request.KeepAlive = false;
-            request.ContentType = "text/html";
-
-            string driverInfomation;
-            AvroDriverInfo info = null;
-            try
+            while (queryUri != null)
             {
-                using (HttpWebResponse webResponse = (HttpWebResponse)request.GetResponse())
+                HttpWebRequest request = (HttpWebRequest)WebRequest.Create(queryUri);
+                request.AllowAutoRedirect = true;
+                request.KeepAlive = false;
+                request.ContentType = "text/html";
+
+                queryUri = null;
+                try
                 {
-                    Stream stream = webResponse.GetResponseStream();
-                    if (stream == null)
-                    {
-                        return null;
-                    }
-                    using (StreamReader streamReader = new StreamReader(stream, Encoding.UTF8))
+                    using (HttpWebResponse webResponse = (HttpWebResponse)request.GetResponse())
                     {
-                        driverInfomation = streamReader.ReadToEnd();
-                        LOGGER.Log(Level.Verbose, "Http response line: " + driverInfomation);
-                        info = AvroJsonSerializer<AvroDriverInfo>.FromString(driverInfomation);
+                        var refresh = webResponse.Headers.AllKeys.FirstOrDefault(k => k.Equals("refresh", StringComparison.OrdinalIgnoreCase));
+
+                        if (refresh != null)
+                        {
+                            var refreshContent = webResponse.Headers.GetValues(refresh);
+                            foreach (var refreshParam in refreshContent.SelectMany(content => content.Split(';').Select(c => c.Trim())))
+                            {
+                                var refreshKeyValue = refreshParam.Split('=').Select(kv => kv.Trim()).ToArray();
+                                if (refreshKeyValue.Length == 2 && refreshKeyValue[0].Equals("url", StringComparison.OrdinalIgnoreCase))
+                                {
+                                    queryUri = new Uri(refreshKeyValue[1]);
+                                    break;
+                                }
+                            }
+                        }
+
+                        // We have received a redirect URI, look there instead.
+                        if (queryUri != null)
+                        {
+                            LOGGER.Log(Level.Verbose, "Received redirect URI:[" + queryUri + "], redirecting...");
+                            Thread.Sleep(500);
```

Comment:
Is there a need to wait here? If it is a redirect, can't we just call that instead immediately?